gtsam/hybrid/tests/testHybridIncremental.cpp

```diff
@@ -0,0 +1,649 @@
+  EXPECT(
+      assert_equal(*x1_conditional, *(expectedHybridBayesNet->atGaussian(0))));
+
+  // The densities on X(2) should be the same
+  auto x2_conditional =
+      dynamic_pointer_cast<GaussianMixture>(isam[X(2)]->conditional()->inner());
+  EXPECT(
+      assert_equal(*x2_conditional, *(expectedHybridBayesNet->atGaussian(1))));
+
+  // // The densities on X(3) should be the same
+  // auto x3_conditional =
+  //     dynamic_pointer_cast<GaussianMixture>(isam[X(3)]->conditional()->inner());
+  // EXPECT(
+  //     assert_equal(*x3_conditional, *(expectedHybridBayesNet->atGaussian(2))));
+
+  GTSAM_PRINT(*expectedHybridBayesNet);
+
+  // we only do the manual continuous elimination for 0,0
+  // the other discrete probabilities on M(2) are calculated the same way
+  auto m00_prob = [&]() {
+    GaussianFactorGraph gf;
+    // gf.add(switching.linearizedFactorGraph.gaussianGraph().at(3));
+
+    DiscreteValues m00;
+    m00[M(1)] = 0, m00[M(2)] = 0;
+    // auto dcMixture =
+    //     dynamic_pointer_cast<DCGaussianMixtureFactor>(graph2.dcGraph().at(0));
+    // gf.add(dcMixture->factors()(m00));
+    // auto x2_mixed =
+    //     boost::dynamic_pointer_cast<GaussianMixture>(hybridBayesNet.at(1));
+    // gf.add(x2_mixed->factors()(m00));
+    auto result_gf = gf.eliminateSequential();
+    return gf.probPrime(result_gf->optimize());
+  }();
+
+  /// Test if the probability values are as expected with regression tests.
+//   DiscreteValues assignment;
+//   EXPECT(assert_equal(m00_prob, 0.60656, 1e-5));
+//   assignment[M(1)] = 0;
+//   assignment[M(2)] = 0;
+//   EXPECT(assert_equal(m00_prob, (*discreteFactor)(assignment), 1e-5));
+//   assignment[M(1)] = 1;
+//   assignment[M(2)] = 0;
+//   EXPECT(assert_equal(0.612477, (*discreteFactor)(assignment), 1e-5));
+//   assignment[M(1)] = 0;
+//   assignment[M(2)] = 1;
+//   EXPECT(assert_equal(0.999952, (*discreteFactor)(assignment), 1e-5));
+//   assignment[M(1)] = 1;
+//   assignment[M(2)] = 1;
+//   EXPECT(assert_equal(1.0, (*discreteFactor)(assignment), 1e-5));
+
+//   DiscreteFactorGraph dfg;
+//   dfg.add(*discreteFactor);
+//   dfg.add(discreteFactor_m1);
+//   dfg.add_factors(switching.linearizedFactorGraph.discreteGraph());
+
+//   // Check if the chordal graph generated from incremental elimination
+//   matches
+//   // that of batch elimination.
+//   auto chordal = dfg.eliminateSequential();
+//   auto expectedChordal =
+//       expectedRemainingGraph->discreteGraph().eliminateSequential();
+
+//   EXPECT(assert_equal(*expectedChordal, *chordal, 1e-6));
+}
+
+/* ****************************************************************************/
+// // Test if we can approximately do the inference
+// TEST(DCGaussianElimination, Approx_inference) {
+//   Switching switching(4);
+//   IncrementalHybrid incrementalHybrid;
+//   HybridGaussianFactorGraph graph1;
+
+//   // Add the 3 DC factors, x1-x2, x2-x3, x3-x4
```

Comment:
These are uncommented in the next PR, right?

Reply:
Yup